drift/lib/src/runtime/types/mapping.dart

```diff
@@ -321,22 +321,40 @@ enum DriftSqlType<T extends Object> implements _InternalDriftSqlType<T> {
     // ignore: unnecessary_cast
     switch (this as DriftSqlType<Object>) {
       case DriftSqlType.bool:
-        return dialect == SqlDialect.sqlite ? 'INTEGER' : 'boolean';
+        return dialect == SqlDialect.sqlite
```

Comment:
Maybe we should make these fields on the `SqlDialect` enum to avoid the chain of ternary expressions here?

Then it could just be `dialect.typeForBool`, for instance.

Reply:
That's a good idea, could you give me some sample code? Not sure where I would put the checks inside of the enum.

Reply:
Ah yeah, that's nice and clean. I will try it tomorrow since it's late now.

By the way, `case DriftSqlType.bigInt:` is empty and goes to `case DriftSqlType.int:`, MariaDB has both `int` and `bigInt`, which one would you recommend we use?

Reply:
We should be aware that should some dialect require a 3rd option for some type, like 
`dialect.booleanType ?? dialect == SqlDialect.mysql? 'smeTypeLiteral': 'someOtherTypeLiteral` we'll probably have to use a ternary expression for it but I think that's ok and probably won't happen since `null` will be rare I think

Reply:
> By the way, case DriftSqlType.bigInt: is empty and goes to case DriftSqlType.int:, MariaDB has both int and bigInt, which one would you recommend we use?

I see how this is a bit confusing, but drift actually only supports one integer type in the database: 64-bit signed integers, which I think is a `bigint` in MariaDb as well. The only difference between `int` and `bigInt` in drift is that the later gets mapped to a `BigInt` in Dart. This can make a difference for web apps, since a Dart `int` on the web is just a double under a trench coat and can't represent large integers properly. But in the database, the expectation is that `DriftSqlType.int` and `DriftSqlType.bigInt` both represent the same type.

>We should be aware that should some dialect require a 3rd option for some type, like
dialect.booleanType ?? dialect == SqlDialect.mysql? 'smeTypeLiteral': 'someOtherTypeLiteral'

Right, but we could just set the `booleanType` property for that dialect then, right?

Reply:
>  but drift actually only supports one integer type in the database: 64-bit signed integers, which I think is a bigint in MariaDb as well

correct, ok so I'll remove the `bigIntType` I added and use `BIGINT` for mariadb got it.

> Right, but we could just set the booleanType property for the affected dialect then, right?

I'm talking about a case of a 2nd different fallback. 
```dart
case DriftSqlType.bool:
  return dialect.booleanType ?? dialect.integerType;
```
we did this, but now imagine some dialect doesn't have a boolean like sqlite, and it uses `dialect.textType` instead, then it would become something like this:
```dart
case DriftSqlType.bool:
  return dialect.booleanType ?? dialect == SqlDialect.someDialect ? dialect.textType : dialect.integerType;
```
This is obviously an extreme example and it doesn't even make sense for bools sepcifically, but could occur for some other type (although I doubt it since the dialects aren't that many, and their differences aren't that great either so I'm just thinking of an extremely unlikely scenario)

Reply:
Yeah, I guess we could cheat and set `booleanType: 'TEXT'` for that dialect. But then, maybe we just do that for sqlite3 as well. Let's make `booleanType` non-nullable and sett it to `INT` for the `sqlite3` dialect.

Reply:
That's a great idea since it will always **have** to be something